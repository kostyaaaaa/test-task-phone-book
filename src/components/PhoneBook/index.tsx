import { useState } from "react";
import { defaultPhoneContacts } from "../../constants.js";
import PhoneContact from "../PhoneContact";
import PhoneNewContact from "../PhoneNewContact";

export type PhoneContactType = {
  name: string;
  phone: string;
  id: number;
};

const PhoneBook = () => {
  const [phoneContacts, setPhoneContacts] =
    useState<PhoneContactType[]>(defaultPhoneContacts);

  const deleteContact = (id: number) => {
    const newPhoneContacts = [...phoneContacts];
    const index = phoneContacts.findIndex(
      (contact: PhoneContactType) => contact.id === id
    );
    newPhoneContacts.splice(index, 1);
    setPhoneContacts(newPhoneContacts);
  };

  const updateContact = ({ name, phone, id }: PhoneContactType) => {
    const newPhoneContacts = [...phoneContacts];
    const index = newPhoneContacts.findIndex((contact) => contact.id === id);
    newPhoneContacts[index] = { name, phone, id };
    setPhoneContacts(newPhoneContacts);
  };

  const addNewContact = (contact: PhoneContactType) => {
    const newPhoneContacts = [...phoneContacts];
    newPhoneContacts.push(contact);
    setPhoneContacts(newPhoneContacts);
  };

  return (
    <div>
      {phoneContacts.map((contact: PhoneContactType) => {
        return (
          <PhoneContact
            updateContact={updateContact}
            deleteContact={deleteContact}
            key={contact.id}
            contact={contact}
          ></PhoneContact>
        );
      })}
      <PhoneNewContact addNewContact={addNewContact} />
    </div>
  );
};

export default PhoneBook;
