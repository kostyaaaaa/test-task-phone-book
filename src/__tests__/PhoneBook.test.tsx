import React from "react";
import PhoneBook from "../components/PhoneBook";
import { shallow } from "enzyme";

let wrapper: any;

describe("PhoneBook", () => {
  beforeEach(() => {
    wrapper = shallow(<PhoneBook></PhoneBook>);
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).not.toBeNull();
  });
});
