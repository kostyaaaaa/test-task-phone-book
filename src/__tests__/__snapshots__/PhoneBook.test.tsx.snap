// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneBook should render correctly 1`] = `
<div>
  <PhoneContact
    contact={
      Object {
        "id": 1,
        "name": "John",
        "phone": "+123456789",
      }
    }
    deleteContact={[Function]}
    key="1"
    updateContact={[Function]}
  />
  <PhoneContact
    contact={
      Object {
        "id": 2,
        "name": "David",
        "phone": "+123123123",
      }
    }
    deleteContact={[Function]}
    key="2"
    updateContact={[Function]}
  />
  <PhoneContact
    contact={
      Object {
        "id": 3,
        "name": "Konstantin",
        "phone": "+098098098",
      }
    }
    deleteContact={[Function]}
    key="3"
    updateContact={[Function]}
  />
  <PhoneContact
    contact={
      Object {
        "id": 4,
        "name": "Eugene",
        "phone": "+12356569",
      }
    }
    deleteContact={[Function]}
    key="4"
    updateContact={[Function]}
  />
  <PhoneNewContact
    addNewContact={[Function]}
  />
</div>
`;
